src/db/sysdb_upgrade.c

```diff
@@ -2236,6 +2236,271 @@ int sysdb_upgrade_17(struct sysdb_ctx *sysdb,
     }
     return ret;
 }
+
+static void add_object_category(struct ldb_context *ldb,
+                                struct upgrade_ctx *ctx)
+{
+    errno_t ret;
+    struct ldb_result *objects = NULL;
+    const char *attrs[] = { SYSDB_OBJECTCLASS, NULL };
+    struct ldb_dn *base_dn;
+    size_t c;
+    const char *class_name;
+    struct ldb_message *msg = NULL;
+
+    base_dn = ldb_dn_new(ctx, ldb, SYSDB_BASE);
+    if (base_dn == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Failed create base dn.\n");
+        return;
+    }
+
+    ret = ldb_search(ldb, ctx, &objects, base_dn,
+                     LDB_SCOPE_SUBTREE, attrs, SYSDB_OBJECTCLASS"=*");
```

Comment:
Do you know if it would be faster to search for objectclasses that are not user or group instead of filtering them from all objects in the for loop?